### ISSUE TITLE
`rules_java` seem to not work with `--nolegacy_external_runfiles` flag

### ISSUE THREAD

**Issue by Otter-man** [bazelbuild/rules_java#203]
The flag docs state:
```# Do not build runfiles symlink forests for external repositories under
# `.runfiles/wsname/external/repo` (in addition to `.runfiles/repo`). This reduces runfiles &
# sandbox creation times & prevents accidentally depending on this feature which may flip to off by
# default in the future. Note, some rules may fail under this flag, please file issues with the rule
# author.
# Docs: https://bazel.build/reference/command-line-reference#flag--legacy_external_runfiles
```

If the flag is added to `.bazelrc`, tests fail with java not found:
```
/private/var/tmp/_bazel_Dan/30d3419be7d4a27db6dff5dc2d690288/sandbox/darwin-sandbox/5542/execroot/_main/bazel-out/darwin_x86_64-fastbuild/bin/plugin-bazel/src/plugin_tests.runfiles/_main/plugin-bazel/src/plugin_tests: line 409: /private/var/tmp/_bazel_Dan/30d3419be7d4a27db6dff5dc2d690288/sandbox/darwin-sandbox/5542/execroot/_main/bazel-out/darwin_x86_64-fastbuild/bin/plugin-bazel/src/plugin_tests.runfiles/_main/external/rules_java~~toolchains~local_jdk/bin/java: No such file or directory
```

**Comment by thirtyseven** [comment]
Do you have a minimal reproduction for this? If it's regarding the `$(JAVABASE)` and `$(JAVA)` Make variables, #272 could address it.
